Leanplum-SDK/Classes/LPRegisterDevice.m

```diff
@@ -51,7 +57,7 @@ - (void)registerDevice:(NSString *)email
         LP_TRY
         BOOL isSuccess = [LPResponse isResponseSuccess:response];
         if (isSuccess) {
-            self->callback(YES);
+            self.callback(YES);
```

Comment:
whats the implications for functionality when we use the dot operator vs the arrow operator? I looked this up, and it seems to indicate: 

>  Arrow operator on an Objective-C object (pointer)

(at run time) Dereference pointer
Return value of the field

Whereas the dot operator has a little bit different functionality...

Reply:
the dot operator basically calls the method `-(type)callback`. If this doesn't exist, it auto-creates one and returns the pointer. So essentially, if we dont have a special need, it'll return the pointer and proceed as above. However, if needed, we could perform any other action (like checking for null), if needed